src/Integration/Aws/SimpleS3/src/SimpleS3Client.php

```diff
@@ -21,8 +26,137 @@ public function getUrl(string $bucket, string $key): string
         return $this->getEndpoint($uri, [], null);
     }
 
-    /*
-     * TODO Add get() and put() which supports large files
-     * TODO add has()
+    public function download(string $bucket, string $key): ResultStream
+    {
+        return $this->getObject(['Bucket' => $bucket, 'Key' => $key])->getBody();
+    }
+
+    public function has(string $bucket, string $key): bool
+    {
+        return $this->objectExists(['Bucket' => $bucket, 'Key' => $key])->isSuccess();
+    }
+
+    /**
+     * @param string|resource|callable|iterable $object
+     * @param array{
+     *   ACL?: \AsyncAws\S3\Enum\ObjectCannedACL::*,
+     *   CacheControl?: string,
+     *   ContentLength?: string,
+     *   ContentType?: string,
+     *   Metadata?: array<string, string>,
+     * } $options
      */
+    public function upload(string $bucket, string $key, $object, array $options = []): void
+    {
+        // Send upload requests with size 32 MB
+        $megabyte = 1024 * 1024;
+        $ceil = 0;
+        if (isset($options['ContentLength'])) {
+            $ceil = ceil((int) $options['ContentLength'] / (10000 * 1024 * 1024));
```

Comment:
hmmm maybe 10000 is a little but too much...

Reply:
I’m not even sure how this works. :/

Reply:
I understand now. 

It increases the part size if the file is around 1TB to make sure we are using less than 10.000 parts.